CMSgov/mint-app
# Fix: The same EUAID will only receive subscription updates on the most recent subscription

# NOREF/ws_pubsub_subscriberid_fix

## Changes and Description

- Websocket pubsub subscription now generates a UUID rather than using the EUAID as a Subscriber ID


## How to test this change

- Start two subscriptions to onLockTaskListSectionContext.
```
{
    "payload": {
        "operationName": "onLockTaskListSectionContext",
        "query": "subscription onLockTaskListSectionContext($modelPlanID: UUID!) { onLockTaskListSectionContext(modelPlanID: $modelPlanID) { changeType lockStatus { modelPlanID section lockedBy } actionType } }",
        "variables": {
            "modelPlanID": "f11eb129-2c80-4080-9440-439cbe1a286f"
        }
    },
    "type": "start"
}
```

- Lock a section on the same model ID.
```
mutation lockTaskListSection($modelPlanID: UUID!, $section: TaskListSection!)
{
    lockTaskListSection(modelPlanID: $modelPlanID, section: $section)
}
```
```
{
  "modelPlanID": "f11eb129-2c80-4080-9440-439cbe1a286f",
  "section": "IT_TOOLS"
}
```

-- Observe that both subscriptions are updated with the lock added status
- Close the second subscription opened
-- Observe that the remaining subscription is updated with the lock removed status

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
